File: 0480-Binary Tree Paths/0480-Binary Tree Paths.cpp
/**
 * Definition of TreeNode:
 * class TreeNode {
 * public:
 *     int val;
 *     TreeNode *left, *right;
 *     TreeNode(int val) {
 *         this->val = val;
 *         this->left = this->right = NULL;
 *     }
 * }
 */
class Solution {
public:
    /**
     * @param root the root of the binary tree
     * @return all root-to-leaf paths
     */
    vector<string> binaryTreePaths(TreeNode* root) {
        // Write your code here
        vector<string> result;
        dfs(root, "", result);
        return result;
    }
private:
    void dfs(TreeNode* root, string path, vector<string>& result)
    {
        if (root == nullptr)
        {
            return;
        }
        
        path += ((path == "") ? "" : "->") + to_string(root->val);
        if (root->left == nullptr && root->right == nullptr)
        {
            result.push_back(path);
        }
        else
        {
            dfs(root->left, path, result);
            dfs(root->right, path, result);
        }
    }
};

File: 0401-0500/0484-Swap Two Integers in Array/0480-Binary Tree Paths.java
public class Solution {
    /**
     * @param A: An integer array
     * @param index1: the first index
     * @param index2: the second index
     * @return: nothing
     */
    public void swapIntegers(int[] A, int index1, int index2) {
        // write your code here
        int temp = A[index1];
        A[index1] = A[index2];
        A[index2] = temp;
    }
}
